### PR TITLE
[infokeydinrail] Initial contribution

### PR DESCRIPTION
**Pull request by thanast**
Infokey Din Rail Binding

This binding is designed to work with some custom made din rail boards based on chips MCP23017, MCP23008 and MCP3008. Also this binding tries to incomporate ready made solution for easy use of cheap sensors that sold in eBay and other networks.

### CODE REVIEW

File: bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	</feature>
+	<feature name="org.openhab.binding.infokeydinrail" description="openHAB Add-ons :: Bundles :: infokeydinrail Binding" version="3.3.0.SNAPSHOT">
+		<details>This project contains the official add-ons of openHAB</details>
+		<feature prerequisite="true" dependency="false">wrap</feature>
+		<bundle start-level="80">mvn:com.pi4j/pi4j-core/1.4</bundle>
+		<bundle start-level="80">mvn:javax.xml.bind/jaxb-api/2.3.1</bundle>
+		<bundle start-level="80">mvn:javax.activation/javax.activation-api/1.2.0</bundle>
+		<bundle start-level="80">wrap:mvn:com.pi4j/pi4j-gpio-extension/1.3</bundle>
+		<bundle start-level="80">wrap:mvn:org.lastnpe.eea/eea-all/2.2.1</bundle>
+	</feature>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml:17]: You don't need this one. Remove it.

Comment (thanast) [reply bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml:17]: I removed it and tried to compile again and the result was the same!

Comment (J-N-K) [reply bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml:17]: Remote `noEmbedDependency.profile`. Why did you add that? It's usually wrong.

Comment (Hilbrand) [reply bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml:17]: @thanast look at the MCP23017 binding to see what you should use in the feature.xml and pom.xml. You don't need to include gson in the pom.xml and also should not use commons io preferably.

**Comment by Hilbrand** [comment]
I've added the work-in-progress label to indicatie this binding needs more work before it should be reviewed (missing readme and lots of commented out code, also misses from the bundle pom.xml).
I was just wondering are these newly developed devies? Looks interesting, but I couldn't find anyrhing in the internet about this.
An additional remark. The pi4j library v1 is deprecated and basically a dead end  imoh (they're developing a new version, but that doesn't seem to have extended support, so I don't know how well it supports ic like the MCP23017 out of the box.) Using this old version 1 might be a problem in the future (If I have it correct the underlying library pi4j depends on is not maintened anymore) and could mean this binding won't work anymore. Just something to consider developing this binding..

File: bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml (hunk 1 above)
Comment (thanast) [reply bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml:17]: > @thanast look at the MCP23017 binding to see what you should use in the feature.xml and pom.xml. You don't need to include gson in the pom.xml and also should not use commons io preferably.

I uploaded this project just for your help. It is not in the phase that can be distributed or can be used. Also I would share the devices in the near future for anyone who will be interested. I have search the web also for issues with karaf and pi4j but nothing found. The intrested thing is that exit another binding called org.opehab.binding.mcp23017 that uses exactly the same libraries do compilation completely to my pc. I tried to copy the project rename it and recompile it compiles successfully. I cannot understand what I missing...

Comment (thanast) [reply bundles/org.openhab.binding.infokeydinrail/src/main/feature/feature.xml:17]: > > @thanast look at the MCP23017 binding to see what you should use in the feature.xml and pom.xml. You don't need to include gson in the pom.xml and also should not use commons io preferably.
> 
> I uploaded this project just for your help. It is not in the phase that can be distributed or can be used. Also I would share the devices in the near future for anyone who will be interested. I have search the web also for issues with karaf and pi4j but nothing found. The intrested thing is that exit another binding called org.opehab.binding.mcp23017 that uses exactly the same libraries do compilation completely to my pc. I tried to copy the project rename it and recompile it compiles successfully. I cannot understand what I missing...

Thank you Hilbrand greetings to The Netherlands from my wife who is Dutch!

File: bundles/org.openhab.binding.infokeydinrail/pom.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.openhab.addons.bundles</groupId>
+    <artifactId>org.openhab.addons.reactor.bundles</artifactId>
+    <version>3.3.0-SNAPSHOT</version>
```
Comment (wborn) [inline bundles/org.openhab.binding.infokeydinrail/pom.xml:10]:
```suggestion
    <version>3.4.0-SNAPSHOT</version>
```

**Comment by lsiepel** [comment]
@thanast are you able to proceed with this PR? If you abandon this PR, it might be better to close it.

**Comment by lsiepel** [comment]
Unfortunately it looks like we cannot bring this PR to the finish line. After two years of inactivity i close this PR. If there is a need to restart, feel free to re-open or start a new PR.

**Pull request closed without merging**